lib/tlscontext.c

```diff
@@ -334,7 +335,25 @@ tls_context_setup_session(TLSContext *self)
         }
 
       SSL_CTX_set_verify(self->ssl_ctx, verify_mode, tls_session_verify_callback);
-      SSL_CTX_set_options(self->ssl_ctx, SSL_OP_NO_SSLv2);
+      if (self->ssl_options != TSO_NONE)
+        {
+          ssl_options=0;
+          if(self->ssl_options & TSO_NOSSLv2)
+            ssl_options |= SSL_OP_NO_SSLv2;
+          if(self->ssl_options & TSO_NOSSLv3)
+            ssl_options |= SSL_OP_NO_SSLv3;
+          if(self->ssl_options & TSO_NOTLSv1)
+            ssl_options |= SSL_OP_NO_TLSv1;
+#ifdef SSL_OP_NO_TLSv1_2
+          if(self->ssl_options & TSO_NOTLSv11)
+            ssl_options |= SSL_OP_NO_TLSv1_1;
+          if(self->ssl_options & TSO_NOTLSv12)
+            ssl_options |= SSL_OP_NO_TLSv1_2;
+#endif
+          SSL_CTX_set_options(self->ssl_ctx, ssl_options);
+        }
+      else
+	msg_debug("empty ssl options",NULL);
```

Comment:
These lines should be extracted into a different function (e.g.: set_ssl_option or something)